From: Benjamin Poirier <bpoirier@suse.de>
Date: Wed, 25 Apr 2012 04:49:20 +0000
Subject: [PATCH] tlan: add cast needed for proper 64 bit operation
Patch-mainline: v3.4-rc6
Git-commit: da3a9e9e7b8341cfc939818103cb7ef9f5293de2
References: bnc#756840

Changes this beauty into a statement that actually has an effect on amd64.

Tested-by: Per Jessen <per@opensuse.org>
Signed-off-by: Benjamin Poirier <bpoirier@suse.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/tlan.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/tlan.c
+++ b/drivers/net/tlan.c
@@ -226,7 +226,7 @@ tlan_get_skb(const struct tlan_list *tag
 	unsigned long addr;
 
 	addr = tag->buffer[9].address;
-	addr |= (tag->buffer[8].address << 16) << 16;
+	addr |= ((unsigned long) tag->buffer[8].address << 16) << 16;
 	return (struct sk_buff *) addr;
 }
 

